src/api/ai_help.rs

```diff
@@ -435,49 +403,130 @@ pub async fn ai_help(
+                                            response_len: i64::try_from(*response_len)
+                                                .unwrap_or(-1),
+                                            model: &model,
+                                            status: (&e).into(),
+                                            sources: Some(&sources_value),
+                                        };
+                                        add_help_message_meta(&mut conn, ai_help_message_meta);
+                                        Some(Err(e))
+                                    }
+                                    None => {
+                                        // This is the added artificial chunk.
+                                        let response_duration = start.elapsed();
+                                        let ai_help_message_meta = AiHelpMessageMetaInsert {
+                                            user_id: user.id,
+                                            chat_id,
+                                            message_id,
+                                            parent_id,
+                                            created_at: Some(created_at.naive_utc()),
+                                            search_duration: i64::try_from(
+                                                ai_help_req_meta.search_duration.as_millis(),
+                                            )
+                                            .unwrap_or(-1),
+                                            response_duration: i64::try_from(
+                                                response_duration.as_millis(),
+                                            )
+                                            .unwrap_or(-1),
+                                            query_len: i64::try_from(ai_help_req_meta.query_len)
+                                                .unwrap_or(-1),
+                                            context_len: i64::try_from(
+                                                ai_help_req_meta.context_len,
+                                            )
+                                            .unwrap_or(-1),
+                                            response_len: i64::try_from(*response_len)
+                                                .unwrap_or(-1),
+                                            model: &model,
+                                            status: db::types::AiHelpMessageStatus::Success,
+                                            sources: Some(&sources_value),
+                                        };
+                                        add_help_message_meta(&mut conn, ai_help_message_meta);
+                                        None
+                                    }
+                                })
+                            }),
+                    ),
+                ))
             }
-            None => {
-                let parts = sorry_response(
```

Comment:
@fiji-flo I don't understand why we can remove the `sorry_response()` call here, but I'm also not sure in which cases we were returning the general "Sorry, I don't know how to help with that." message.